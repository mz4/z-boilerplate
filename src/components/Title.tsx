import React, { useState } from "react";
import "./Title.scss";

// https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example

interface IProps {
  players: {
    id: number;
    name: string;
    role: string;
    age: number;
    active: boolean;
  }[];
  deletePlayer: (id: number) => void;
}

const Title = ({ players, deletePlayer }: IProps) => {
  const [num, setNum] = useState<number>(0);

  const changeNum = (): void => {
    setNum(num + 1);
  };

  const renderList = (): JSX.Element[] => {
    return players.map((player, index) => {
      return (
        <tr key={index}>
          <td>{player.id}</td>
          <td>{player.name}</td>
          <td>{player.role}</td>
          <td>{player.age}</td>
          <td>{player.active ? "yes" : "no"}</td>
          <td>
            <button
              data-testid="deleteplayer"
              onClick={(event: React.MouseEvent<HTMLElement>) =>
                deletePlayer(player.id)
              }
            >
              Delete
            </button>
          </td>
        </tr>
      );
    });
  };

  return (
    <div>
      <h2>This is a title</h2>
      <button onClick={changeNum}>
        Increase Number <span>{num}</span>
      </button>
      <table>
        <thead>
          <tr>
            <th>Id</th>
            <th>Name</th>
            <th>Role</th>
            <th>Age</th>
            <th>Active</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody>{renderList()}</tbody>
      </table>
    </div>
  );
};

export default Title;
