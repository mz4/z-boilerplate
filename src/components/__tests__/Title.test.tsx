import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react";
import Title from "../Title";

const propertiesV1 = {
  players: [
    {
      id: 1,
      name: "Lebron James",
      role: "Shooting Guard",
      age: 36,
      active: true,
    },
  ],
  deletePlayer: jest.fn(),
};

describe("Title", () => {
  it("should match the snapshot", () => {
    const { container } = render(<Title {...propertiesV1} />);
    expect(container).toMatchSnapshot();
  });
  it("should find queries by text", () => {
    const { queryByText } = render(<Title {...propertiesV1} />);
    expect(queryByText("This is a title")).toBeInTheDocument();
    expect(queryByText("Increase Number")).toBeInTheDocument();
    expect(queryByText("Id")).toBeInTheDocument();
    expect(queryByText("Name")).toBeInTheDocument();
    expect(queryByText("Role")).toBeInTheDocument();
    expect(queryByText("Age")).toBeInTheDocument();
    expect(queryByText("Active")).toBeInTheDocument();
    expect(queryByText("Action")).toBeInTheDocument();
  });
  it("should call delete player", () => {
    const { getByTestId } = render(<Title {...propertiesV1} />);
    fireEvent.click(getByTestId("deleteplayer"));
    expect(propertiesV1.deletePlayer).toHaveBeenCalled();
  });
});
