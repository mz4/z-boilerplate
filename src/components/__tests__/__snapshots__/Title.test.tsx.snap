// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match the snapshot 1`] = `
<div>
  <div>
    <h2>
      This is a title
    </h2>
    <button>
      Increase Number 
      <span>
        0
      </span>
    </button>
    <table>
      <thead>
        <tr>
          <th>
            Id
          </th>
          <th>
            Name
          </th>
          <th>
            Role
          </th>
          <th>
            Age
          </th>
          <th>
            Active
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            Lebron James
          </td>
          <td>
            Shooting Guard
          </td>
          <td>
            36
          </td>
          <td>
            yes
          </td>
          <td>
            <button
              data-testid="deleteplayer"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
